executors/kv/kvexecutor.go

```diff
@@ -0,0 +1,136 @@
+package executor
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sort"
+	"strings"
+	"sync"
+	"time"
+)
+
+// KVExecutor is a simple in-memory key-value store that implements the Executor interface for testing purposes.
```

Comment:
nit: if this is only for testing purposes, should it be `test/executors/kv` for clarity?

Reply:
otherwise lgtm

Reply:
makes sense